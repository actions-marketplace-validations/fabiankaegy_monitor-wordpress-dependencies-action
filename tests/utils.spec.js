const { toBool, diffTable, fileExists, parseAssetFile } = require('../lib/utils.js');
const path = require('path');

test('toBool', () => {
	expect(toBool('1')).toBe(true);
	expect(toBool('true')).toBe(true);
	expect(toBool('yes')).toBe(true);

	expect(toBool('0')).toBe(false);
	expect(toBool('false')).toBe(false);
	expect(toBool('no')).toBe(false);
});

test('diffTable', () => {
	const files = [
		{
			filename: 'sample/index.asset.php',
			dependency: ['wp-block-editor', 'wp-blocks', 'wp-element', 'wp-primitives',],
			added: ['wp-primitives'],
			removed: ['wp-components'],
		  },
		  {
			filename: 'sample/test.asset.php',
			dependency: ['wp-block-editor', 'wp-blocks', 'wp-element', 'wp-primitives',],
			added: ['wp-primitives, wp-components'],
			removed: [''],
		  },
		  {
			filename: 'sample/hello.asset.php',
			dependency: ['wp-block-editor', 'wp-blocks', 'wp-element', 'wp-primitives'],
			added: ['wp-primitives',],
			removed: [''],
		  },
		  {
			filename: 'sample/view.asset.php',
			dependency: ['wp-block-editor', 'wp-blocks', 'wp-element', 'wp-primitives'],
			added: ['wp-primitives',],
			removed: [''],
		  }
	];
	const defaultOptions = {
		showTotal: true,
		collapseUnchanged: true,
		omitUnchanged: false,
	};

	expect(diffTable(files, { ...defaultOptions })).toMatchSnapshot();
	expect(diffTable(files, { ...defaultOptions, showTotal: false })).toMatchSnapshot();
	expect(diffTable(files, { ...defaultOptions, collapseUnchanged: false })).toMatchSnapshot();
	expect(diffTable(files, { ...defaultOptions, omitUnchanged: true })).toMatchSnapshot();

	expect(diffTable([files[2]], { ...defaultOptions })).toMatchSnapshot();
});

test('fileExists', async () => {
	expect(await fileExists('package.json')).toBe(true);
	expect(await fileExists('file-that-does-not-exist')).toBe(false);
});

test('parseAssetFile', async () => {
	const assetFixturePath = path.resolve( './fixtures/after/index.asset.php' );
	expect(await parseAssetFile(assetFixturePath)).toMatchSnapshot();
});
