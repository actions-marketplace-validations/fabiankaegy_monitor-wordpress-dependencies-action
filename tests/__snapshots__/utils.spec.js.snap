// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`diffTable 1`] = `
"| Filename | Dependencies | Change |  |
| :--- | :---: | :---: | :---: |
| \`sample/index.asset.php\` | wp-block-editor, wp-blocks, wp-element, wp-primitives | wp-primitives | wp-components |
| \`sample/test.asset.php\` | wp-block-editor, wp-blocks, wp-element, wp-primitives | wp-primitives, wp-components |  |
| \`sample/hello.asset.php\` | wp-block-editor, wp-blocks, wp-element, wp-primitives | wp-primitives |  |
| \`sample/view.asset.php\` | wp-block-editor, wp-blocks, wp-element, wp-primitives | wp-primitives |  |"
`;

exports[`diffTable 2`] = `
"| Filename | Dependencies | Change |  |
| :--- | :---: | :---: | :---: |
| \`sample/index.asset.php\` | wp-block-editor, wp-blocks, wp-element, wp-primitives | wp-primitives | wp-components |
| \`sample/test.asset.php\` | wp-block-editor, wp-blocks, wp-element, wp-primitives | wp-primitives, wp-components |  |
| \`sample/hello.asset.php\` | wp-block-editor, wp-blocks, wp-element, wp-primitives | wp-primitives |  |
| \`sample/view.asset.php\` | wp-block-editor, wp-blocks, wp-element, wp-primitives | wp-primitives |  |"
`;

exports[`diffTable 3`] = `
"| Filename | Dependencies | Change |  |
| :--- | :---: | :---: | :---: |
| \`sample/index.asset.php\` | wp-block-editor, wp-blocks, wp-element, wp-primitives | wp-primitives | wp-components |
| \`sample/test.asset.php\` | wp-block-editor, wp-blocks, wp-element, wp-primitives | wp-primitives, wp-components |  |
| \`sample/hello.asset.php\` | wp-block-editor, wp-blocks, wp-element, wp-primitives | wp-primitives |  |
| \`sample/view.asset.php\` | wp-block-editor, wp-blocks, wp-element, wp-primitives | wp-primitives |  |"
`;

exports[`diffTable 4`] = `
"| Filename | Dependencies | Change |  |
| :--- | :---: | :---: | :---: |
| \`sample/index.asset.php\` | wp-block-editor, wp-blocks, wp-element, wp-primitives | wp-primitives | wp-components |
| \`sample/test.asset.php\` | wp-block-editor, wp-blocks, wp-element, wp-primitives | wp-primitives, wp-components |  |
| \`sample/hello.asset.php\` | wp-block-editor, wp-blocks, wp-element, wp-primitives | wp-primitives |  |
| \`sample/view.asset.php\` | wp-block-editor, wp-blocks, wp-element, wp-primitives | wp-primitives |  |"
`;

exports[`diffTable 5`] = `
"| Filename | Dependencies | Change |
| :--- | :---: | :---: |
| \`sample/hello.asset.php\` | wp-block-editor, wp-blocks, wp-element, wp-primitives | wp-primitives |"
`;

exports[`parseAssetFile 1`] = `
{
  "dependencies": [
    "wp-block-editor",
    "wp-blocks",
    "wp-components",
    "wp-element",
  ],
  "version": "176fde31d1a79ed7ca30",
}
`;
